File: Problem-046/solution.py
def goldbachs_other_conjecture():

    n = 9

    while True:

        # Check only composite numbers
        if not prime(n):

            found = False

            # Try every prime smaller than n
            for p in range(2, n):

                if prime(p):

                    temp = n - p

                    # Check whether temp = 2 × square
                    if temp % 2 == 0 and int((temp / 2) ** 0.5) ** 2 == temp / 2:
                        found = True
                        break

            # If no representation exists
            if not found:
                return n

        # Move to the next odd number
        n += 2


def prime(n):

    if n < 2:
        return False

    i = 2

    while i * i <= n:

        if n % i == 0:
            return False

        i += 1

    return True


print(goldbachs_other_conjecture())

File: Problem-046/solution.js
function goldbachsOtherConjecture() {

    let n = 9;

    while (true) {

        if (!prime(n)) {

            let k = false;

            for (let p = 2; p < n; p++) {

                if (prime(p)) {

                    let temp = n - p;

                    if (
                        temp % 2 === 0 &&
                        Number.isInteger(Math.sqrt(temp / 2))
                    ) {
                        k = true;
                        break;
                    }
                }
            }

            if (!k) {
                return n;
            }
        }

        n += 2;
    }
}


function prime(n) {

    if (n < 2) return false;

    for (let i = 2; i * i <= n; i++) {

        if (n % i === 0) {
            return false;
        }
    }

    return true;
}


console.log(goldbachsOtherConjecture());
